subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -556,34 +587,12 @@ static void hawkbit_dump_deployment(struct hawkbit_dep_res *d)
 int hawkbit_init(void)
 {
 	bool image_ok;
-	int ret = 0, rc = 0;
-	struct flash_pages_info info;
-	int32_t action_id;
-
-	fs.flash_device = STORAGE_DEV;
-	if (!device_is_ready(fs.flash_device)) {
-		LOG_ERR("Flash device not ready");
-		return -ENODEV;
-	}
-
-	fs.offset = STORAGE_OFFSET;
-	rc = flash_get_page_info_by_offs(fs.flash_device, fs.offset, &info);
-	if (rc) {
-		LOG_ERR("Failed to get storage page info: %d", rc);
-		return -EIO;
-	}
-
-	fs.sector_size = info.size;
-	fs.sector_count = 3U;
+	int ret = 0;
 
-	rc = nvs_mount(&fs);
-	if (rc) {
-		LOG_ERR("Failed to mount storage flash: %d", rc);
-		return rc;
-	}
+	settings_subsys_init();
+	settings_load_subtree("hawkbit");
```

Comment:
what would happen if setting fails to init/load?

Reply:
now handeled